src/http/proxy.mbt

```diff
@@ -0,0 +1,100 @@
+// Copyright 2025 International Digital Economy Academy
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+///|
+/// HTTP proxy configuration
+enum Proxy {
+  ForHttp(String, Int)
+  ForHttps(String, Int)
+  ForAll(String, Int)
+  Custom(http~ : (String, Int), https~ : (String, Int))
+}
```

Comment:
I am not sure if this abstraction is necessary. For `@http.Client::connect` and `@http.get`, the protocol is always known. So protocol based proxy selection should be done manually by the user IMO, and `@http` should just accept an optional `(host, port)` pair for proxy